File: Lv.3/추석 트래픽.js
function solution(lines) {
    let answer = 0;
    let start = [];
    let end = [];
    for(let i=0;i<lines.length;i++){
        let h,m,s,ms;
        let ih,im,is,dot;
        
        
        h = lines[i].substring(11);
        m = lines[i].substring(14);
        s = lines[i].substring(17);
        ms = lines[i].substring(20);
        dot = parseFloat(lines[i].substring(24))*1000;
        
        ih = parseInt(h)*3600*1000;
        im = parseInt(m)*60*1000;
        is = parseInt(s)*1000+parseInt(ms);
        
        start.push(ih+im+is-dot+1);
        end.push(ih+im+is);
    }
    
    for(let i=0;i<lines.length;i++){
        let end_time = end[i]+1000;
        let cnt = 0;
        for(let j=i;j<lines.length;j++){
            if(start[j] < end_time){
                cnt += 1;
            }
        }
        if(answer < cnt){
            answer = cnt;
        }
    }
    
    return answer;
}

File: Lv.3/추석 트래픽.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

int solution(vector<string> lines) {
    int answer = 0;
    vector<int> start;
    vector<int> end;
    for(int i=0;i<lines.size();i++){
        string h,m,s,ms;
        int ih,im,is,dot;
        cout<<lines[i]<<' ';
        lines[i].pop_back();
        cout<<lines[i]<<' ';
        h = lines[i].substr(11,2);
        m = lines[i].substr(14,2);
        s = lines[i].substr(17,2);
        ms = lines[i].substr(20,3);
        dot = stof(lines[i].substr(24,5))*1000;
        
        ih = stoi(h)*3600*1000;
        im = stoi(m)*60*1000;
        is = stoi(s)*1000+stoi(ms);
        start.push_back(ih+im+is-dot+1);
        end.push_back(ih+im+is);
    }
    for(int i=0;i<lines.size();i++){
        int end_time = end[i]+1000;
        int cnt=0;
        for(int j=i;j<lines.size();j++){
            if(start[j]<end_time){
                cnt++;
            }
        }
        answer = max(answer,cnt);
    }
    return answer;
}